packages/gatsby/index.d.ts

```diff
@@ -932,32 +933,58 @@ export interface Store {
   replaceReducer: Function
 }
 
-type logMessageType = (format: string, ...args: any[]) => void
-type logErrorType = (message: string, error?: Error) => void
+type LogMessageType = (format: string, ...args: any[]) => void
+type LogErrorType = (errorMeta: string | Object, error?: Object) => void
+
+export type PhantomActivityTracker = {
+  start(): () => void
+  end(): () => void
+  span: Object
+}
+
+export type ActivityTracker = PhantomActivityTracker & {
+  setStatus(status: string): void
+  panic: LogErrorType
+  panicOnBuild: LogErrorType
+}
+
+export type ProgressActivityTracker = Omit<ActivityTracker, "end"> & {
+  tick(increment?: number): void
+  done(): void
+  total: number
+}
+
+export type ActivityArgs = {
+  parentSpan?: Object
+  id?: string
+}
 
 export interface Reporter {
-  stripIndent: Function
+  stripIndent: TemplateTag
   format: object
-  setVerbose(isVerbose: boolean): void
-  setNoColor(isNoColor: boolean): void
-  panic: logErrorType
-  panicOnBuild: logErrorType
-  error: logErrorType
+  setVerbose(isVerbose?: boolean): void
+  setNoColor(isNoColor?: boolean): void
+  panic: LogErrorType
+  panicOnBuild: LogErrorType
+  error: LogErrorType
   uptime(prefix: string): void
-  success: logMessageType
-  verbose: logMessageType
-  info: logMessageType
-  warn: logMessageType
-  log: logMessageType
-  activityTimer(
-    name: string,
-    activityArgs: { parentSpan: object }
-  ): {
-    start: () => void
-    status(status: string): void
-    end: () => void
-    span: object
-  }
+  success: LogMessageType
+  verbose: LogMessageType
+  info: LogMessageType
+  warn: LogMessageType
+  log: LogMessageType
+  completeActivity(id: string, status?: string): void
```

Comment:
I'm not sure we want  `completeActivity` in public type definitions - this is more for internal `gatsby` usage and I would vote for not adding typing for that

Reply:
thanks, yeah I did wonder about that, will remove.